### ISSUE TITLE
Improve feature loading performance by changing UUID generator

### ISSUE THREAD

**Issue by jkronegg** [cucumber/cucumber-expressions#213]
### 👓 What did you see?

The Cucumber Java event bus uses UUID as event identifiers. These UUIDs are generated using the JDK `java.util.UUID.randomUUID()`.
This kind of generators known to be slow (e.g. https://stackoverflow.com/questions/14532976/performance-of-random-uuid-generation-with-java-7-or-java-6). On my project (~450 test scenarios, ~50 rules) this UUID generator takes about 25% of the feature file loading time on the IntelliJ profiler flame graph (~2% of the total CPU time).

### ✅ What did you expect to see?

Using a faster UUID generator would improve the feature file loading performance. For example, https://github.com/cowtowncoder/java-uuid-generator provides a time sequence UUID generator which is much faster. 

From what I've seen, the solution could be very simple as the `EventBus` is instanciated in `io.cucumber.core.runtime.Runtime.Builder` with `UUID::randomUUID` : we need to change this generator to get a good performance improvement. Note that other calls to `UUID.randomUUID()` are also made elsewhere in the project but I didn't saw big performance impact.

### 📦 Which tool/library version are you using?

Cucumber Java 7.11.1

### 🔬 How could we reproduce it?

1. Run IntelliJ profiler on a project with about 50 rules with cucumber-java 7.11.1.
2. Look for the `loadFeatures` block on the flame graph: on the top is the `java.util.UUID.randomUUID()` call

### 📚 Any additional context?

N/A

**Comment by jkronegg** [comment]
The following librairies have been selected as alternatives to generate the UUIDs:

| Library                                                   | Licence             | Benchmarked by creator                                                  | Latest commit                                                                                          |
|-----------------------------------------------------------|---------------------|-------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------|
| JDK java.util.UUID.randomUUID() (OpenJDK 17)              | ?                   | ?                                                                       | [may 2022](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/UUID.java) |
| https://github.com/cowtowncoder/java-uuid-generator (JUG) | Apache License 2.0. | [yes](http://www.cowtowncoder.com/blog/archives/2010/10/entry_429.html) | jan 2023                                                                                               |
| https://github.com/f4b6a3/uuid-creator                    | MIT License         | [yes](https://github.com/f4b6a3/uuid-creator/wiki/5.0.-Benchmark)       | oct 2022                                                                                               |

As Cucumber has a MIT License project, all three alternatives are allowed [from the license point of view](https://opensource.stackexchange.com/questions/11640/can-mit-and-apache-licenses-be-used-together).

The following libraries have not been selected as potential replacements:
- The [Apache commons-id](https://commons.apache.org/sandbox/commons-id/uuid.html) library was not selected because it looks unmaintened (latest build in 2006) and [has no official release](https://commons.apache.org/sandbox/commons-id/downloads.html) (it's still in sandbox mode).
- The [eaio-uuid](https://github.com/stephenc/eaio-uuid) library was not selected because it looks unmaintened (latest commit in 2018) and is not documented. Moreover, it generates it's own `com.eaoi.uuid.UUID` class, not the JVM `java.util.UUID`, which would make integration harder.

Regarding thread-safety, all selected alternatives are thread-safe:
- [uuid-creator is benchmarked with multiple threads](https://github.com/f4b6a3/uuid-creator/wiki/5.0.-Benchmark), so it is assumed to be thread-safe
- [JUG is described as thread-safe](https://github.com/cowtowncoder/java-uuid-generator) (I didn't verify)

I benchmarked the UUID librairies with some parameters that are known to provide better performance. 
 
I also tested the generators in realworld conditions on a project with ~450 test scenarios and ~50 rules, I passes a `Supplier<UUID>` instance which measures the time passed to generate the UUIDs:

```java
public class UuidSuplier implements Supplier<UUID> {
  static AtomicLong time = new AtomicLong(0);

    @Override
    public UUID get() {
        long t0 = System.nanoTime();
        UUID uuid = ... // generate the UUID
        long t1 = System.nanoTime();
        System.out.println(time.addAndGet(t1-t0)); // displays total time in ns
        return uuid;
    }
}
```

These are the results:

| Library     | generator type              | JMH benchmark Millions ops/s (higher is better) | real-world performance; time to generate all UUIDs [ms] (lower is better) |
|-------------|-----------------------------|-------------------------------------------------|---------------------------------------------------------------------------|
| JUG         | time-based epoch (Type 7)   | 1.1                                             | 268                                                                       |
| JDK         | random (Type 4)             | 1.1                                             | 189                                                                       |
| uuidCreator | time-based (Type 1)         | 10.0                                            | 99                                                                        |
| uuidCreator | time-based epoch (Type 7/2) | 34.5                                            | 77                                                                        |
| JUG         | time-based (Type 1)         | too fast to be benchmarked                      | 12                                                                        |

To summarise, by using the JUG library with time-based Type 1 UUID generator, the gain is about 250 ms on a real project (about 10% of the Cucumber test scenario execution on the project).

@mpkorstanje,  @mattwynne : are you ok with the principle of replacing `java.util.UUID.randomUuid()` by a JUG time-based UUID ? If yes, I'll propose a PR.
